File: python/1380.py
class Solution:
    def luckyNumbers (self, matrix: list[list[int]]) -> list[int]:
        row_min = [min(row) for row in matrix]
        col_max = [max(col) for col in zip(*matrix)]
        return list(set(row_min) & set(col_max))
       
if __name__ == '__main__':
    n, m = int(input()), int(input())
    matrix = [list(map(int, input().split())) for _ in range(n)]
    s = Solution()
    print(s.luckyNumbers(matrix)) 

File: cpp/1380.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> ans;
        for(int i=0; i<matrix.size(); i++){
            int min = matrix[i][0], idx = 0;
            for(int j=1; j<matrix[i].size(); j++){
                if(matrix[i][j] < min){
                    min = matrix[i][j];
                    idx = j;
                }
            }
            bool flag = true;
            for(int k=0; k<matrix.size(); k++){
                if(matrix[k][idx] > min){
                    flag = false;
                    break;
                }
            }
            if(flag) ans.push_back(min);
        }
        return ans;
    }
};

int main(){
    int m, n;
    cin >> m >> n;
    vector<vector<int>> matrix(m, vector<int>(n));
    for(int i=0; i<m; i++){
        for(int j=0; j<n; j++){
            cin >> matrix[i][j];
        }
    }
    Solution s;
    vector<int> ans = s.luckyNumbers(matrix);
    for(int i=0; i<ans.size(); i++){
        cout << ans[i] << " ";
    }
    cout << endl;
    return 0;
}